### PR TITLE
Add health tracking for ipfs

### PR DESCRIPTION
**Pull request by pjz**
This adds one variable 'avail' that's the available space in the datastore.  It then uses that and the previously built 'size' variable to add an alert when the datastore gets over 90% full.

My only question is this: parsing the config isn't something that should be done every call, so I just parse it the first time and then cache it, so that costly work is only done once.  Server config isn't likely to change very often, so I thought it'd be acceptable to require a netdata restart if it did.  Should I instead figure out how to parse the config at some interval  (once every 5 or 10 mins maybe)?

Oh, and I also switched the size graph to `area` instead of `line` because that's what other disk-space-related graphs use.

### CODE REVIEW

**Comment by ktsaou** [comment]
I think this is ok. But, since I don't use ipfs, let's invite the initial contributor of this module...

@davidak do you have any comments?

(also note that I have already merged another PR of @pjz to this)

**Comment by davidak** [comment]
looks good for me. havn't tested it.

File: conf.d/health.d/ipfs.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+   alarm: ipfs_datastore_in_use
+      on: ipfs_local.repo_size
```
Comment (ktsaou) [inline conf.d/health.d/ipfs.conf]: I suggest to change this to `template` in `ipfs.repo_size`, so that it will be applied to all ipfs instances.

**Comment by ktsaou** [comment]
@davidak thanks!

@pjz just one suggestion to make the alarm match all ipfs instances. Could you please update it?

File: conf.d/health.d/ipfs.conf (hunk 1 above)
Comment (pjz) [inline conf.d/health.d/ipfs.conf]: To be clear (I'm a bit new to netdata alarms), I need to change
`on: ipfs_local.repo_size` to be `template: ipfs_local.repo_size` ?

**Comment by ktsaou** [comment]
@pjz you also need to remove `_local` from the `on` line for the template to work. Templates work on the `context` of charts, not the ids.

**Comment by pjz** [comment]
better?

**Comment by ktsaou** [comment]
perfect!

**Comment by davidak** [comment]
@pjz thanks for the contribution!

**Comment by pjz** [comment]
You're welcome! Scratching my own itch, really. Thanks for making netdata so awesome!

**Pull request merged**